packages/indexer-common/src/allocations/query-fees.ts

```diff
@@ -367,14 +410,177 @@ export class AllocationReceiptCollector implements ReceiptCollector {
+              chainId: this.protocolNetwork.split(':')[1],
+              name: 'TAP',
+              version: '1',
+              verifyingContract: this.escrowContracts!.tapVerifier.address,
+            }
+            const signedRav = rav.getSignedRAV()
+            console.log({ domain, rav })
+            const signerAddress = utils.verifyTypedData(
+              domain,
+              {
+                ReceiptAggregateVoucher: [
+                  { name: 'allocationId', type: 'address' },
+                  { name: 'timestampNs', type: 'uint64' },
+                  { name: 'valueAggregate', type: 'uint128' },
+                ],
+              },
+              signedRav.rav,
+              signedRav.signature,
+            )
+            console.log({ signerAddress, ravSigner: rav.senderAddress })
+
+            return {
+              rav: signedRav,
+              allocation: allocations.find(
+                (a) => a.id === toAddress(signedRav.rav.allocationId),
+              ),
+              sender: rav.senderAddress,
+            }
+          })
+          .filter((rav) => rav.allocation !== undefined) as RavWithAllocation[] // this is safe because we filter out undefined allocations
+      },
+      { onError: (err) => this.logger.error(`Failed to query pending RAVs`, { err }) },
+    )
+  }
+
+  protected getSignedRAVsEventual(
+    pendingRAVs: Eventual<RavWithAllocation[]>,
+  ): Eventual<ValidRavs> {
+    return pendingRAVs.tryMap(
+      async (pendingRAVs) => {
+        return await pReduce(
+          pendingRAVs,
+          async (results, rav) => {
+            // allocationId doesn't have 0x prefix, but still matched in contract calls
+            // destory voucher if AllocationIDTracker.isAllocationIdUsed(rav.allocationId)
+            if (
+              BigNumber.from(rav.rav.rav.valueAggregate).lt(
+                this.voucherRedemptionThreshold,
+              )
+            ) {
+              results.belowThreshold.push(rav)
+            } else {
+              results.eligible.push(rav)
+            }
+            return results
+          },
+          { belowThreshold: <RavWithAllocation[]>[], eligible: <RavWithAllocation[]>[] },
+        )
+      },
+      { onError: (err) => this.logger.error(`Failed to reduce to signed RAVs`, { err }) },
+    )
+  }
+
+  // redeem only if last is true
+  // Later can add order and limit
+  protected async pendingRAVs(): Promise<ReceiptAggregateVoucher[]> {
+    let ravsToRedeem: ReceiptAggregateVoucher[] = []
+    const unfinalizedRAVs = await this.models.receiptAggregateVouchers.findAll({
+      where: { last: true, final: false },
+    })
+    // Obtain allocationIds to use as filter in subgraph
+    const allocationIdsAddress = unfinalizedRAVs.map(
+      (rav) => rav.getSignedRAV().rav.allocationId.toLowerCase(),
+    )
+    const allocationIds = unfinalizedRAVs.map(
+      (rav) => rav.getSignedRAV().rav.allocationId.toLowerCase().replace('0x', ''),
+    )
```

Comment:
This can be removed

Reply:
The reason why its done is because in the subgraph is the full address and in the db is without the `0x`, this caused the query not to find those allocations

Reply:
You don't need this `allocationIds` if you use the same where clause in your update.

Reply:
I did tried it and it does not work, it should be an exact match per object in the array

Reply:
```suggestion

    if (allocationIdsAddress.length > 0) {
      console.log(`Get on-chain redeemed RAVs from TAP subgraph`)
```